libs/server/AOF/AofProcessor.cs

```diff
@@ -259,40 +259,113 @@ private unsafe bool ReplayOp(byte* entryPtr)
                     ObjectStoreDelete(objectStoreBasicContext, entryPtr);
                     break;
                 case AofEntryType.StoredProcedure:
-                    ref var input = ref Unsafe.AsRef<SpanByte>(entryPtr + sizeof(AofHeader));
-                    respServerSession.RunTransactionProc(header.type, new ArgSlice(ref input), ref output);
+                    RunStoredProc(header.type, entryPtr);
                     break;
                 default:
                     throw new GarnetException($"Unknown AOF header operation type {header.opType}");
             }
             return true;
         }
 
-        static unsafe void StoreUpsert(BasicContext<SpanByte, SpanByte, SpanByte, SpanByteAndMemory, long, MainSessionFunctions, MainStoreFunctions, MainStoreAllocator> basicContext, byte* ptr)
+        unsafe void RunStoredProc(byte id, byte* ptr)
         {
-            ref var key = ref Unsafe.AsRef<SpanByte>(ptr + sizeof(AofHeader));
-            ref var input = ref Unsafe.AsRef<SpanByte>(ptr + sizeof(AofHeader) + key.TotalSize);
-            ref var value = ref Unsafe.AsRef<SpanByte>(ptr + sizeof(AofHeader) + key.TotalSize + input.TotalSize);
+            var curr = ptr;
+            var parseStateCount = *(int*)curr;
+            curr += sizeof(int);
+
+            var parseState = new SessionParseState();
```

Comment:
The AOF can keep a single parseState allocated at constructor time, similar to RespServerSession, so that we do not incur an allocation for each of the million calls into RunStoredProc.